lib/components/ArtistScreen/album_list_tile.dart

```diff
@@ -0,0 +1,583 @@
+      /// song in an album.
+      this.index,
+      this.parentId,
+      this.parentName,
+      this.onDelete})
+      : super(key: key);
+
+  final BaseItemDto item;
+  final List<BaseItemDto>? children;
+  final int? index;
+  final String? parentId;
+  final String? parentName;
+  final VoidCallback? onDelete;
+
+  @override
+  State<AlbumListTile> createState() => _AlbumListTileState();
+}
+
+class _AlbumListTileState extends State<AlbumListTile> {
+  final _audioServiceHelper = GetIt.instance<AudioServiceHelper>();
+  final _queueService = GetIt.instance<QueueService>();
+  final _audioHandler = GetIt.instance<MusicPlayerBackgroundTask>();
+  final _jellyfinApiHelper = GetIt.instance<JellyfinApiHelper>();
+
+  @override
+  Widget build(BuildContext context) {
+    final screenSize = MediaQuery.of(context).size;
+
+    final listTile = ListTile(
+      leading: AlbumImage(item: widget.item),
+      title: RichText(
+        text: TextSpan(
+          children: [
+            // third condition checks if the item is viewed from its album (instead of e.g. a playlist)
+            // same horrible check as in canGoToAlbum in GestureDetector below
+            TextSpan(
+              text:
+                  widget.item.name ?? AppLocalizations.of(context)!.unknownName,
+            ),
+          ],
+          style: Theme.of(context).textTheme.titleMedium,
+        ),
+      ),
+      subtitle: RichText(
+        text: TextSpan(
+          children: [
+            WidgetSpan(
+              child: Transform.translate(
+                offset: const Offset(-3, 0),
+                child: DownloadedIndicator(
+                  item: widget.item,
+                  size: Theme.of(context).textTheme.bodyMedium!.fontSize! + 3,
+                ),
+              ),
+              alignment: PlaceholderAlignment.top,
+            ),
+            TextSpan(
+              text: widget.item.productionYearString,
+              style: TextStyle(
+                  color: Theme.of(context)
+                      .textTheme
+                      .bodyMedium
+                      ?.color
+                      ?.withOpacity(0.7)),
+            ),
+            TextSpan(
+              text:
+                  " · ${printDuration(Duration(microseconds: (widget.item.runTimeTicks == null ? 0 : widget.item.runTimeTicks! ~/ 10)))}",
+              style: TextStyle(color: Theme.of(context).disabledColor),
+            ),
+          ],
+        ),
+        overflow: TextOverflow.ellipsis,
+      ),
+      trailing: FavoriteButton(
+        item: widget.item,
+        onlyIfFav: true,
+      ),
+      onTap: () {
+        if (widget.item.type == "MusicArtist" ||
+            widget.item.type == "MusicGenre") {
+          Navigator.of(context)
+              .pushNamed(ArtistScreen.routeName, arguments: widget.item);
+        } else {
+          Navigator.of(context)
+              .pushNamed(AlbumScreen.routeName, arguments: widget.item);
+        }
+      },
```

Comment:
This is an **album** list tile, and should only support the case of the parent being an album. Trying to handle all cases in one place it what got us into this mess in the first place :D

Reply:
Woops, pebkac I forgot what I was copying from